### PR TITLE
vimc-4595 Add dependencies endpoint

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Transforms the dependency graph result from orderly into digestible json. 

The `dependency_tree` property at the top level of the `Dependencies` response is a single `Dependency` node which represents the root report for which dependencies are being returned. The `dependencies` property recursively contains the dependencies for that report. 

Where there are no dependencies an empty array is returned in `dependencies`.

### CODE REVIEW

File: tests/testthat/test-z-integration.R [hunk 1]
```diff
@@ -448,3 +448,18 @@ test_that("Can create a report with parameters", {
     readRDS(file.path(tmp, id, "meta", "info.rds"))$parameters,
     list(time = 10, poll = 1))
 })
+
+test_that("Can get dependencies", {
+  server <- start_test_server()
```
Comment (r-ash) [inline tests/testthat/test-z-integration.R:453]: Should we test with a report with more dependencies than `count` report?

You should be able to start test server with more reports by using
```
path <- orderly_git_example("demo")
server <- start_test_server(path)
```
which has reports, `use_dependency` or `use_dependency_2`

File: R/api.R [hunk 2]
```diff
@@ -307,6 +307,37 @@ endpoint_kill <- function(runner) {
     returning = returning_json("Kill.schema"))
 }
 
+target_dependencies <- function(path, name,
+                                id ="latest",
```
Comment (r-ash) [inline R/api.R:311]:
```suggestion
                                id = "latest",
```

File: tests/testthat/test-dependencies.R [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  expect_equal(r3$name, scalar("r3"))
+  expect_equal(r3$id, scalar("3"))
+  expect_equal(r3$out_of_date, scalar(FALSE))
+  expect_equal(length(r3$dependencies), 1)
+  r4 <- r3$dependencies[[1]]
+  expect_equal(r4$name, scalar("r4"))
+  expect_equal(r4$id, scalar("4"))
+  expect_equal(r4$out_of_date, scalar(TRUE))
+  expect_equal(length(r4$dependencies), 0)
+})
```
Comment (r-ash) [inline tests/testthat/test-dependencies.R:79]:
```suggestion
})

```

**Review by r-ash — APPROVED** [review]
Couple of tiny format comments but this looks great

File: tests/testthat/test-z-integration.R (hunk 1 above)
Comment (r-ash) [reply tests/testthat/test-z-integration.R:453]: Actually just seen `test-api-endpoints.R` covers this case.

File: R/api.R (hunk 2 above)
Comment (EmmaLRussell) [reply R/api.R]: Surprised the linter didn't pick this up, it moaned about everything else! :laughing:

**Pull request merged**